modules/core/src/v2/coins/eos.ts

```
@@ -0,0 +1,219 @@
+      prv: extendedKey.toBase58(),
+    };
+  };
+
+  /**
+   * Return boolean indicating whether input is valid public key for the coin.
+   *
+   * @param pub - the pub to be checked
+   */
+  isValidPub(pub: string): boolean {
+    try {
+      utxoLib.HDNode.fromBase58(pub);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   * Return boolean indicating whether input is valid seed for the coin
+   *
+   * @param prv - the prv to be checked
+   */
+  isValidPrv(prv: string): boolean {
+    try {
+      utxoLib.HDNode.fromBase58(prv);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   * Evaluates whether a memo is valid
+   *
+   * @param memo - the memo to be checked
+   */
+  isValidMemo = function(memo: string): boolean {
+    return memo && memo.length <= 256;
+  };
+
+  /**
+   * Return boolean indicating whether a memo id is valid
+   *
+   * @param memoId - the memo id to be checked
+   */
+  isValidMemoId = function(memoId: string): boolean {
+    if (!this.isValidMemo(memoId)) {
+      return false;
+    }
+
+    let memoIdNumber;
+    try {
+      memoIdNumber = new BigNumber(memoId);
+    } catch (e) {
+      return false;
+    }
+
+    return memoIdNumber.gte(0);
+  };
+
+  /**
+   * Process address into address and memo id
+   * @param address - the address
+   */
+  getAddressDetails(address: string): AddressDetails {
+    const destinationDetails = url.parse(address);
+    const queryDetails = querystring.parse(destinationDetails.query);
+    const destinationAddress = destinationDetails.pathname;
+
+    // EOS addresses have to be "human readable", which means start with a letter, less than 12 characters and only a-z1-5., i.e.mtoda1.bitgo
+    // source: https://developers.eos.io/eosio-cpp/docs/naming-conventions
+    if (!/^[a-z][a-z1-5.]*$/.test(destinationAddress) || destinationAddress.length > EOS_ADDRESS_LENGTH) {
+      throw new Error(`invalid address: ${address}`);
```

Should throw an `InvalidAddressError` since we already have that error type defined